sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceivedMessage.java

```diff
@@ -17,49 +29,65 @@
  */
 public final class ServiceBusReceivedMessage {
+    private final ClientLogger logger = new ClientLogger(ServiceBusReceivedMessage.class);
+
+    private static final String DEAD_LETTER_DESCRIPTION = "DeadLetterErrorDescription";
+    private static final String DEAD_LETTER_REASON = "DeadLetterReason";
+
+    // This one appears to always be 0, but is always returned with each message.
+    private static final String ENQUEUED_SEQUENCE_NUMBER = "x-opt-enqueue-sequence-number";
+
+    private final AmqpAnnotatedMessage amqpAnnotatedMessage;
     private UUID lockToken;
-    private long sequenceNumber;
-    private long enqueuedSequenceNumber;
-    private long deliveryCount;
-    private OffsetDateTime enqueuedTime;
-    private OffsetDateTime lockedUntil;
-    private String deadLetterSource;
-
-    private final Map<String, Object> properties;
-    private final byte[] body;
-    private String contentType;
-    private String correlationId;
-    private String label;
-    private String messageId;
-    private String partitionKey;
-    private String replyTo;
-    private String replyToSessionId;
-    private OffsetDateTime scheduledEnqueueTime;
-    private String sessionId;
-    private Duration timeToLive;
-    private String to;
-    private String viaPartitionKey;
-    private String deadLetterReason;
-    private String deadLetterErrorDescription;
+
+    /**
+     * The representation of message as defined by AMQP protocol.
+     *
+     * @see <a href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#section-message-format">
+     *     Amqp Message Format.</a>
+     *
+     * @return the {@link AmqpAnnotatedMessage} representing amqp message.
+     */
+    public AmqpAnnotatedMessage getAmqpAnnotatedMessage() {
+        return amqpAnnotatedMessage;
+    }
 
     ServiceBusReceivedMessage(byte[] body) {
-        this.body = Objects.requireNonNull(body, "'body' cannot be null.");
-        this.properties = new HashMap<>();
+        Objects.requireNonNull(body, "'body' cannot be null.");
+        amqpAnnotatedMessage = new AmqpAnnotatedMessage(new AmqpDataBody(new BinaryData(body)));
     }
 
     /**
      * Gets the actual payload/data wrapped by the {@link ServiceBusReceivedMessage}.
      *
      * <p>
      * If the means for deserializing the raw data is not apparent to consumers, a common technique is to make use of
-     * {@link #getProperties()} when creating the event, to associate serialization hints as an aid to consumers who
-     * wish to deserialize the binary data.
+     * {@link #getApplicationProperties()} ()} when creating the event, to associate serialization hints as an aid to
+     * consumers who wish to deserialize the binary data.
      * </p>
      *
      * @return A byte array representing the data.
      */
     public byte[] getBody() {
-        return Arrays.copyOf(body, body.length);
+        byte[] body = null;
```

Comment:
The point I was trying to make is that you know this data in the constructor. You can do this logic in the constructor and simply return a copy of the binary data. 
Before, you were iterating through the Iterable of BinaryData each time, creating a new List object and then throwing it away to get the first item before copying it.